File: sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/TestCases/SwitchCasesMinimumThree.cs
namespace Tests.Diagnostics
{
    public class SwitchCasesMinimumThree
    {
        public SwitchCasesMinimumThree(int n)
        {
            switch (n) // Noncompliant
//          ^^^^^^
            {
                case 0:
                    break;
                default:
                    break;
            }

            switch (n) // Noncompliant {{Replace this 'switch' statement with 'if' statements to increase readability.}}
            {
            }

            switch (n)
            {
                case 0:
                    break;
                case 1:
                default:
                    var x=5;
                    break;
            }
        }
    }
}

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SwitchCasesMinimumThree.vb
Namespace Tests.Diagnostics
    Public Class SwitchCasesMinimumThree
        Public Sub New(ByVal n As Integer)
            Select Case n ' Noncompliant {{Replace this 'Select' statement with 'If' statements to increase readability.}}
'           ^^^^^^
                Case 0
                Case Else
            End Select

            Select Case n ' Noncompliant {{Replace this 'Select' statement with 'If' statements to increase readability.}}
'           ^^^^^^
            End Select

            Select Case n 'Compliant, 3 cases
                Case 0, 1
                Case Else
                    Dim x = 5
            End Select
        End Sub
    End Class
End Namespace

